core/src/main/scala/io/finch/request/package.scala

```diff
@@ -309,7 +358,9 @@ package object request extends LowPriorityImplicits {
    *
    * @param message the message
    */
-  class RequestReaderError(val message: String) extends Exception(message)
+  class RequestReaderError(val message: String, cause: Throwable) extends Exception(message, cause) {
+    def this(message: String) = this(message, null)
```

Comment:
That's a pretty imperfect disguise, I'd say. :-D

I don't see that as an actual improvement, but I also totally would not mind doing that. I'd recommend to keep `cause` non-optional in `NotParsed` though, as we always have a cause there, so it would only be wrapped in a `Some` for calling the super constructor.